nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -144,18 +146,17 @@ LifecycleManager::destroyLifecycleServiceClients()
 bool
 LifecycleManager::createBondConnection(const std::string & node_name)
 {
-  const double timeout_s = bond_timeout_.count();
+  const double timeout_ns =
+    std::chrono::duration_cast<std::chrono::nanoseconds>(bond_timeout_).count();
+  const double timeout_s = timeout_ns / 1e9;
 
-  if (bond_map_.find(node_name) == bond_map_.end() && timeout_s > 0.0) {
+  if (bond_map_.find(node_name) == bond_map_.end() && bond_timeout_.count() > 0.0) {
     bond_map_[node_name] =
       std::make_shared<bond::Bond>("bond", node_name, bond_client_node_);
     bond_map_[node_name]->setHeartbeatTimeout(timeout_s);
     bond_map_[node_name]->setHeartbeatPeriod(0.10);
     bond_map_[node_name]->start();
-    if (
-      !bond_map_[node_name]->waitUntilFormed(
-        rclcpp::Duration(rclcpp::Duration::from_seconds(timeout_s / 2))))
-    {
+    if (!bond_map_[node_name]->waitUntilFormed(rclcpp::Duration(timeout_ns / 2))) {
```

Comment:
This API is deprecated, you cannot use Duration without a conversion or chrono literal